quack/rmsnorm.py

```diff
@@ -773,3 +809,39 @@ def rmsnorm(x: torch.Tensor, weight: torch.Tensor, eps: float = 1e-6) -> torch.T
         Normalized output tensor of same shape as x
     """
     return RMSNormFunction.apply(x, weight, eps)
+
+
+class QuackRMSNorm(torch.nn.Module):
```

Comment:
we can just subclass torch.nn.RMSNorm instead? and only change the forward?

Reply:
in theory yes, but I started trying this tonight and hit various cute dsl issues in the process.
Thus, I'd like to push this off as a seperate PR rather than try and do it here as we already have a lot of changes with bf16/fp16 support coming in. 
subclassing may also give us free torch.compile support as that's a different problem atm (we have to provide a dummy wrapper for compile to not try and dive into the quack code). 
Opened Issue https://github.com/Dao-AILab/quack/issues/18 to track torch.compile issue (and by proxy, rmsnorm subclassing)